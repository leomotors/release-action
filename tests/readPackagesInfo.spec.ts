import { describe, expect, it } from "vitest";

import {
  getPackageNameAndVersion,
  safeReadPackagesInfo,
} from "../src/modules/utils/readPackagesInfo";

describe("safeReadPackagesInfo", () => {
  it("Valid Case", async () => {
    expect(
      await safeReadPackagesInfo("tests/__fixtures__/pkg-info-success-1.yaml"),
    ).toMatchSnapshot();
  });

  it("Parse Fail Case", async () => {
    expect(
      await safeReadPackagesInfo("tests/__fixtures__/pkg-info-fail-1.yaml"),
    ).toMatchSnapshot();
  });

  it("Invalid File Case", async () => {
    expect(
      await safeReadPackagesInfo("tests/__fixtures__/pkg-info-fail-1.yml"),
    ).toMatchSnapshot();
  });
});

describe("getPackageNameAndVersion", () => {
  it("Normal Case", () => {
    expect(getPackageNameAndVersion("1.2.3")).toStrictEqual({
      packageName: "",
      version: "1.2.3",
    });

    expect(getPackageNameAndVersion("1.2.3-rc.1")).toStrictEqual({
      packageName: "",
      version: "1.2.3-rc.1",
    });

    expect(getPackageNameAndVersion("package@1.2.3")).toStrictEqual({
      packageName: "package",
      version: "1.2.3",
    });

    expect(getPackageNameAndVersion("package@1.2.3-rc.1")).toStrictEqual({
      packageName: "package",
      version: "1.2.3-rc.1",
    });

    expect(getPackageNameAndVersion("package/path@1.2.3")).toStrictEqual({
      packageName: "package/path",
      version: "1.2.3",
    });

    expect(getPackageNameAndVersion("package/path/nest@1.2.3")).toStrictEqual({
      packageName: "package/path/nest",
      version: "1.2.3",
    });

    expect(getPackageNameAndVersion("@package/path@1.2.3")).toStrictEqual({
      packageName: "@package/path",
      version: "1.2.3",
    });

    expect(getPackageNameAndVersion("@package/path/nest@1.2.3")).toStrictEqual({
      packageName: "@package/path/nest",
      version: "1.2.3",
    });
  });
});
