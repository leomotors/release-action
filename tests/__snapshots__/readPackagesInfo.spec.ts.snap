// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`safeReadPackagesInfo > Invalid File Case 1`] = `
{
  "error": "Failed to read packages info file: Error: ENOENT: no such file or directory, open 'tests/__fixtures__/pkg-info-fail-1.yml'",
  "success": false,
}
`;

exports[`safeReadPackagesInfo > Parse Fail Case 1`] = `
{
  "error": [ZodError: [
  {
    "code": "invalid_type",
    "expected": "array",
    "received": "undefined",
    "path": [
      "packages"
    ],
    "message": "Required"
  }
]],
  "success": false,
}
`;

exports[`safeReadPackagesInfo > Valid Case 1`] = `
{
  "data": {
    "packages": [
      {
        "fullName": "Hello Package",
        "name": "hello",
        "packagePath": "apps/hello",
      },
      {
        "name": "world",
        "packagePath": "apps/world",
      },
      {
        "name": "@with/changelog",
        "packagePath": "packages/with-changelog",
      },
      {
        "name": "apps/something",
        "packagePath": "apps/something",
      },
    ],
  },
  "success": true,
}
`;
